### PR TITLE
Added SoundTouch to be able to adjust pitch and tempo with effecting the audio with side effects.

### PR DESCRIPTION
**Pull request by ArizonaInkStudios**

### CODE REVIEW

**Comment by filoe** [comment]
Great! But could you please move it to a seperate project within the Samples folder? Would like to keep main CSCore project clean from dependencies. Similar to the https://github.com/filoe/cscore/tree/master/Samples/NVorbisIntegration sample.

**Comment by ArizonaInkStudios** [comment]
Sure thing!

**Comment by ArizonaInkStudios** [comment]
I moved it.

File: CSCore/CSCore.csproj [hunk 1]
```diff
@@ -619,10 +619,12 @@
   <ItemGroup>
     <Service Include="{508349B6-6B84-4DF5-91F0-309BEEBAD82D}" />
   </ItemGroup>
+  <ItemGroup>
```
Comment (filoe) [inline CSCore/CSCore.csproj]: CSCore.csproj still contains SoundTouch.dll. Also the Libs/SoundTouch.dll file still exists within CSCore.

File: CSCore.sln [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2013
-VisualStudioVersion = 12.0.40629.0
+# Visual Studio 15
```
Comment (filoe) [inline CSCore.sln]: Not sure whether this will break support for older versions of visual studio. Maybe we can switch it back? @ArizonaInkStudios

**Comment by filoe** [comment]
Great, I've added two comments. Maybe we can fix that and merge it afterwards.
Thanks!

**Comment by ArizonaInkStudios** [comment]
All complete. Let me know if we have any other oddities.

Did you give the sample a go?

**Comment by filoe** [comment]
Sample looks good to me. 
Just one question. Why is there a coefficient of 8? 
See https://github.com/ArizonaInkStudios/cscore/blob/master/SoundTouchPitchAndTempo/SoundTouchSource.cs#L74 and following. 

Also one last please: Move the sample to the Samples folder.
Thanks! :+1.

**Comment by ArizonaInkStudios** [comment]
The coefficient of 8 is based on my music knowledge and audio pitch calculations. 8 equals a half semaphore which is half a step on pitch. That could possibly not be correct for this, but it does work. If you have any suggestions I am definitely open to something different. Anything to make the code cleaner and more precise I am up for!

**Comment by filoe** [comment]
I don't think that this will really work. There are different input pcm formats like 8, 16, 24, 32 bit. Also the underlying WaveSource might provide Ieeefloat. 
I would suggest the following:
1. Inherit from `SampleAggregatorBase` instead of `WaveAggregatorBase`
2. The `Read` method will provide floating point values automatically. 
3. If SoundTouch provides more samples than you can read, you would have to handle that differently. Before putting new samples to SoundTouch you would have to make sure that you already received all samples. 
4. Have you thought about seeking? When the position gets changed, you would have to reset the SoundTouch buffer (flush or what ever - don't know SoundTouch). Also does SoundTouch change the number samples of a track by processing them? Let's say you've got 100 input samples. Is the output always 100 samples? If not, is there a way for handling that when seeking?

**Comment by ArizonaInkStudios** [comment]
Those are things to consider absolutely. I do have a separate project with this code working and it seeks without issue; it does work. Nonetheless, perhaps more testing with other PCM formats is in order to ensure that works.

Most of the tests I have done are with MP3 that are 16 but I believe. I will create some test wave files and do some hard testing to see what may break or cause issues.

I will take your suggestions and see what changes can be made. You know your project better than anyone. Thank you for your help! Hopefully we can get a good sample working that fulfills all the sources.

**Comment by ArizonaInkStudios** [comment]
Here is my results from testing.

I used:
8bits-48khz.wav
16bits-48khz.wav
24bits-48khz.wav
32bits-IEEE-48khz.wav

All of these worked with the code as is.

I also did seeking and position changing while tempo/pitch change and it all worked as expected.

I tried to convert to SampleAggregatorBase and use float buffers as the base and that was bad; VERY bad. Using the float buffer in many different attempts I either got an exception and bad explosions or the audio was choppy, had static, and was not audio that was good in any way. I am all for updating this but I am a bit stumped on how to accomplish it. I am open for suggestions.

Sorry this is such an ordeal. I will not give up. If there is a better answer, I would like to find it.

**Comment by filoe** [comment]
Sorry for the late response. I've merged it. Thanks!

**Comment by ArizonaInkStudios** [comment]
No worries! Glad I could help and contribute something. I hope to do more in the future! Great working with you.

**Pull request merged**